pkg/backend/openshiftcluster.go

```diff
@@ -89,13 +89,16 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 	case api.ProvisioningStateCreating:
 		log.Print("creating")
 
-		err = m.Create(ctx)
+		ok, err := m.Create(ctx)
```

Comment:
I am worrying about some potential nastiness here.  Create() messes around with the model, but the "doc" at this level is presumably stale at the point where Create() exits.  If so I am worried we then use the stale doc below in updateAsyncOperation?

I think we should probably re-Get the doc after calling m.Create() (and after m.Update() as well, for good measure).

Returning `ok` here is pretty ugly.  Having done the above, you can then inspect 			`doc.OpenShiftCluster.Properties.Install` at this point: if nil, then `return ocb.endLease(ctx, stop, doc, api.ProvisioningStateSucceeded)`.  If non-nil, then `return ocb.endLease(ctx, stop, doc, api.ProvisioningStateCreating)`.

Reply:
Hmmm. I was thinking about this implementation. Will try and see how it looks